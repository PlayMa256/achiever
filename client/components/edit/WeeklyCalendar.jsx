import React from 'react';
import PropTypes from 'prop-types';
import BigCalendar from 'react-big-calendar';
import moment from 'moment';

import 'react-big-calendar/lib/css/react-big-calendar.css';

import strings from '../../../shared/strings';

import '../../styles/weeklyCalendar.styl';

BigCalendar.setLocalizer(BigCalendar.momentLocalizer(moment));

const noOp = () => () => {};

const _getDateFromComposedObj = (dateObj, entryType) => {
	const dateFormatSent = 'YYYY-MM-DD H:mm';
	return moment(`${dateObj.date} ${dateObj[entryType]}`, dateFormatSent).toDate();
};

const _convertweekEntriesToEvents = (weekEntries) => {
	const emptyValidReturn = [{}];
	if (weekEntries.timeEntries) {
		const events = [];
		[0, 1, 2, 3, 4, 5, 6].forEach((index) => {
			const dayEntry = weekEntries.timeEntries[index];
			if (dayEntry) {
				const hasBreak = dayEntry.startBreakTime && dayEntry.endBreakTime;
				if (hasBreak) {
					events.push({
						id: index * 2,
						start: _getDateFromComposedObj(dayEntry, 'startTime'),
						end: _getDateFromComposedObj(dayEntry, 'startBreakTime'),
						title: `${dayEntry.phase} - ${dayEntry.activity}`
					});
					events.push({
						id: (index * 2) + 1,
						start: _getDateFromComposedObj(dayEntry, 'endBreakTime'),
						end: _getDateFromComposedObj(dayEntry, 'endTime'),
						title: `${dayEntry.phase} - ${dayEntry.activity}`
					});
				} else {
					events.push({
						id: index * 2,
						start: _getDateFromComposedObj(dayEntry, 'startTime'),
						end: _getDateFromComposedObj(dayEntry, 'endTime'),
						title: `${dayEntry.phase} - ${dayEntry.activity}`
					});
				}
			}
		});

		return events || emptyValidReturn;
	}
	return emptyValidReturn;
};

const WeeklyCalendar = ({ weekEntries, controlDate }) => (
	<details className="weekly-calendar">
		<summary>{strings.weeklyCalendar}</summary>
		<BigCalendar
			view="week"
			onView={noOp}
			step={90}
			onNavigate={noOp}
			date={controlDate.toDate()}
			toolbar={false}
			selectable={false}
			events={_convertweekEntriesToEvents(weekEntries)}
		/>
	</details>
);

WeeklyCalendar.propTypes = {
	controlDate: PropTypes.object,
	weekEntries: PropTypes.object
};

WeeklyCalendar.defaultProps = {
	controlDate: {},
	weekEntries: {}
};

export default WeeklyCalendar;
