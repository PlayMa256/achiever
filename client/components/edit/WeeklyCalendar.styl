@import "../../styles/_colours.styl"
@import "../../styles/_standards.styl"

.weekly-calendar
    clear: both;
    width: 80%;
    margin: 0 auto;
    padding: 20px 0;
    height: 350px;

.rbc
    &-time-header
        .rbc-row:nth-of-type(2) // hack to not to show the "all-day" row
            display: none

    &-day-slot, &-header
        margin: 0 2px;

    &-label
        font-size: 12px;

    &-day-slot &-event
        border: 0;
        &-label
            color: rgba(255,255,255, .5);
            font-size: 10px;
            margin-bottom: 2px;
        &-content
            color: rgba(255,255,255, .7);
            font-size: 10px;

    &-today
        background-color: white;