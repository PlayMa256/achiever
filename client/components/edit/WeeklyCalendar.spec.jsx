import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment-timezone';

import WeeklyCalendar from './WeeklyCalendar';

const defaultProps = {
	controlDate: moment('2018-02-14T00:00:00.000Z'),
	weekEntries: {
		timeEntries: {
			0: {
				date: '2018-02-11',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '11:35',
				endBreakTime: '12:35',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			1: {
				date: '2018-02-12',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '11:35',
				endBreakTime: '12:35',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			2: {
				date: '2018-02-13',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '11:35',
				endBreakTime: '12:35',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			3: {
				date: '2018-02-14',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '',
				endBreakTime: '',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			4: {
				date: '2018-02-15',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '11:35',
				endBreakTime: '12:35',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			5: {
				date: '2018-02-16',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '',
				endBreakTime: '',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			},
			6: {
				date: '2018-02-17',
				phase: 'Project Phase Name',
				activity: 'Code/ Document Reviews',
				startTime: '9:28',
				startBreakTime: '11:35',
				endBreakTime: '12:35',
				endTime: '18:28',
				total: '8:00',
				__typename: 'TimeEntry'
			}
		},
		total: '56:00'
	},
	storedTimes: [
		{ hours: 7, minutes: 0 },
		{ hours: 12, minutes: 0 },
		{},
		{}
	]
};

describe('WeeklyCalendar', () => {
	describe('render', () => {
		it('should render the BigCalendar dealing with the props', () => {
			const wrapper = shallow(<WeeklyCalendar
				controlDate={defaultProps.controlDate}
				weekEntries={defaultProps.weekEntries}
				storedTimes={defaultProps.storedTimes}
			/>);
			const emptyWrapper = shallow(<WeeklyCalendar
				controlDate={defaultProps.controlDate}
				weekEntries={{ timeEntries: null }}
				storedTimes={defaultProps.storedTimes}
			/>);
			expect(wrapper).toMatchSnapshot();
			expect(emptyWrapper).toMatchSnapshot();
		});
	});
});
