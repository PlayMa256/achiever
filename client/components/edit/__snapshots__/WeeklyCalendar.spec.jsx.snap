// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyCalendar render should render the BigCalendar dealing with the props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeeklyCalendar
    controlDate={"2018-02-14T02:00:00.000Z"}
    storedTimes={
      Array [
        Object {
          "hours": 7,
          "minutes": 0,
        },
        Object {
          "hours": 12,
          "minutes": 0,
        },
        Object {},
        Object {},
      ]
    }
    weekEntries={
      Object {
        "timeEntries": Object {
          "0": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-11",
            "endBreakTime": "12:35",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "11:35",
            "startTime": "9:28",
            "total": "8:00",
          },
          "1": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-12",
            "endBreakTime": "12:35",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "11:35",
            "startTime": "9:28",
            "total": "8:00",
          },
          "2": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-13",
            "endBreakTime": "12:35",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "11:35",
            "startTime": "9:28",
            "total": "8:00",
          },
          "3": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-14",
            "endBreakTime": "",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "",
            "startTime": "9:28",
            "total": "8:00",
          },
          "4": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-15",
            "endBreakTime": "12:35",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "11:35",
            "startTime": "9:28",
            "total": "8:00",
          },
          "5": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-16",
            "endBreakTime": "",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "",
            "startTime": "9:28",
            "total": "8:00",
          },
          "6": Object {
            "__typename": "TimeEntry",
            "activity": "Code/ Document Reviews",
            "date": "2018-02-17",
            "endBreakTime": "12:35",
            "endTime": "18:28",
            "phase": "Project Phase Name",
            "startBreakTime": "11:35",
            "startTime": "9:28",
            "total": "8:00",
          },
        },
        "total": "56:00",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <summary>
          Calendário da semana
        </summary>,
        <Uncontrolled(Calendar)
          date={2018-02-14T02:00:00.000Z}
          events={
            Array [
              Object {
                "end": 2018-02-11T13:35:00.000Z,
                "id": 0,
                "start": 2018-02-11T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-11T20:28:00.000Z,
                "id": 1,
                "start": 2018-02-11T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-12T13:35:00.000Z,
                "id": 2,
                "start": 2018-02-12T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-12T20:28:00.000Z,
                "id": 3,
                "start": 2018-02-12T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-13T13:35:00.000Z,
                "id": 4,
                "start": 2018-02-13T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-13T20:28:00.000Z,
                "id": 5,
                "start": 2018-02-13T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-14T14:00:00.000Z,
                "id": 6,
                "start": 2018-02-14T09:00:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-14T02:00:00.000Z,
                "id": 7,
                "start": 2018-02-14T02:00:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-15T13:35:00.000Z,
                "id": 8,
                "start": 2018-02-15T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-15T20:28:00.000Z,
                "id": 9,
                "start": 2018-02-15T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-16T20:28:00.000Z,
                "id": 10,
                "start": 2018-02-16T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-17T13:35:00.000Z,
                "id": 12,
                "start": 2018-02-17T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-17T20:28:00.000Z,
                "id": 13,
                "start": 2018-02-17T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
            ]
          }
          onNavigate={[Function]}
          onView={[Function]}
          selectable={false}
          step={90}
          toolbar={false}
          view="week"
        />,
      ],
      "className": "weekly-calendar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Calendário da semana",
        },
        "ref": null,
        "rendered": "Calendário da semana",
        "type": "summary",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "date": 2018-02-14T02:00:00.000Z,
          "events": Array [
            Object {
              "end": 2018-02-11T13:35:00.000Z,
              "id": 0,
              "start": 2018-02-11T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-11T20:28:00.000Z,
              "id": 1,
              "start": 2018-02-11T14:35:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-12T13:35:00.000Z,
              "id": 2,
              "start": 2018-02-12T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-12T20:28:00.000Z,
              "id": 3,
              "start": 2018-02-12T14:35:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-13T13:35:00.000Z,
              "id": 4,
              "start": 2018-02-13T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-13T20:28:00.000Z,
              "id": 5,
              "start": 2018-02-13T14:35:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-14T14:00:00.000Z,
              "id": 6,
              "start": 2018-02-14T09:00:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-14T02:00:00.000Z,
              "id": 7,
              "start": 2018-02-14T02:00:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-15T13:35:00.000Z,
              "id": 8,
              "start": 2018-02-15T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-15T20:28:00.000Z,
              "id": 9,
              "start": 2018-02-15T14:35:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-16T20:28:00.000Z,
              "id": 10,
              "start": 2018-02-16T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-17T13:35:00.000Z,
              "id": 12,
              "start": 2018-02-17T11:28:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
            Object {
              "end": 2018-02-17T20:28:00.000Z,
              "id": 13,
              "start": 2018-02-17T14:35:00.000Z,
              "title": "Project Phase Name - Code/ Document Reviews",
            },
          ],
          "onNavigate": [Function],
          "onView": [Function],
          "selectable": false,
          "step": 90,
          "toolbar": false,
          "view": "week",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "details",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <summary>
            Calendário da semana
          </summary>,
          <Uncontrolled(Calendar)
            date={2018-02-14T02:00:00.000Z}
            events={
              Array [
                Object {
                  "end": 2018-02-11T13:35:00.000Z,
                  "id": 0,
                  "start": 2018-02-11T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-11T20:28:00.000Z,
                  "id": 1,
                  "start": 2018-02-11T14:35:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-12T13:35:00.000Z,
                  "id": 2,
                  "start": 2018-02-12T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-12T20:28:00.000Z,
                  "id": 3,
                  "start": 2018-02-12T14:35:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-13T13:35:00.000Z,
                  "id": 4,
                  "start": 2018-02-13T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-13T20:28:00.000Z,
                  "id": 5,
                  "start": 2018-02-13T14:35:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-14T14:00:00.000Z,
                  "id": 6,
                  "start": 2018-02-14T09:00:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-14T02:00:00.000Z,
                  "id": 7,
                  "start": 2018-02-14T02:00:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-15T13:35:00.000Z,
                  "id": 8,
                  "start": 2018-02-15T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-15T20:28:00.000Z,
                  "id": 9,
                  "start": 2018-02-15T14:35:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-16T20:28:00.000Z,
                  "id": 10,
                  "start": 2018-02-16T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-17T13:35:00.000Z,
                  "id": 12,
                  "start": 2018-02-17T11:28:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
                Object {
                  "end": 2018-02-17T20:28:00.000Z,
                  "id": 13,
                  "start": 2018-02-17T14:35:00.000Z,
                  "title": "Project Phase Name - Code/ Document Reviews",
                },
              ]
            }
            onNavigate={[Function]}
            onView={[Function]}
            selectable={false}
            step={90}
            toolbar={false}
            view="week"
          />,
        ],
        "className": "weekly-calendar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calendário da semana",
          },
          "ref": null,
          "rendered": "Calendário da semana",
          "type": "summary",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "date": 2018-02-14T02:00:00.000Z,
            "events": Array [
              Object {
                "end": 2018-02-11T13:35:00.000Z,
                "id": 0,
                "start": 2018-02-11T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-11T20:28:00.000Z,
                "id": 1,
                "start": 2018-02-11T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-12T13:35:00.000Z,
                "id": 2,
                "start": 2018-02-12T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-12T20:28:00.000Z,
                "id": 3,
                "start": 2018-02-12T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-13T13:35:00.000Z,
                "id": 4,
                "start": 2018-02-13T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-13T20:28:00.000Z,
                "id": 5,
                "start": 2018-02-13T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-14T14:00:00.000Z,
                "id": 6,
                "start": 2018-02-14T09:00:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-14T02:00:00.000Z,
                "id": 7,
                "start": 2018-02-14T02:00:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-15T13:35:00.000Z,
                "id": 8,
                "start": 2018-02-15T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-15T20:28:00.000Z,
                "id": 9,
                "start": 2018-02-15T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-16T20:28:00.000Z,
                "id": 10,
                "start": 2018-02-16T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-17T13:35:00.000Z,
                "id": 12,
                "start": 2018-02-17T11:28:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
              Object {
                "end": 2018-02-17T20:28:00.000Z,
                "id": 13,
                "start": 2018-02-17T14:35:00.000Z,
                "title": "Project Phase Name - Code/ Document Reviews",
              },
            ],
            "onNavigate": [Function],
            "onView": [Function],
            "selectable": false,
            "step": 90,
            "toolbar": false,
            "view": "week",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "details",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WeeklyCalendar render should render the BigCalendar dealing with the props 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeeklyCalendar
    controlDate={"2018-02-14T02:00:00.000Z"}
    storedTimes={
      Array [
        Object {
          "hours": 7,
          "minutes": 0,
        },
        Object {
          "hours": 12,
          "minutes": 0,
        },
        Object {},
        Object {},
      ]
    }
    weekEntries={
      Object {
        "timeEntries": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <summary>
          Calendário da semana
        </summary>,
        <Uncontrolled(Calendar)
          date={2018-02-14T02:00:00.000Z}
          events={
            Array [
              Object {},
            ]
          }
          onNavigate={[Function]}
          onView={[Function]}
          selectable={false}
          step={90}
          toolbar={false}
          view="week"
        />,
      ],
      "className": "weekly-calendar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Calendário da semana",
        },
        "ref": null,
        "rendered": "Calendário da semana",
        "type": "summary",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "date": 2018-02-14T02:00:00.000Z,
          "events": Array [
            Object {},
          ],
          "onNavigate": [Function],
          "onView": [Function],
          "selectable": false,
          "step": 90,
          "toolbar": false,
          "view": "week",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "details",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <summary>
            Calendário da semana
          </summary>,
          <Uncontrolled(Calendar)
            date={2018-02-14T02:00:00.000Z}
            events={
              Array [
                Object {},
              ]
            }
            onNavigate={[Function]}
            onView={[Function]}
            selectable={false}
            step={90}
            toolbar={false}
            view="week"
          />,
        ],
        "className": "weekly-calendar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calendário da semana",
          },
          "ref": null,
          "rendered": "Calendário da semana",
          "type": "summary",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "date": 2018-02-14T02:00:00.000Z,
            "events": Array [
              Object {},
            ],
            "onNavigate": [Function],
            "onView": [Function],
            "selectable": false,
            "step": 90,
            "toolbar": false,
            "view": "week",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "details",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
